atom/browser/browser_win.cc

```diff
@@ -215,13 +215,21 @@ bool Browser::SetAsDefaultProtocolClient(const std::string& protocol,
 
 bool Browser::IsDefaultProtocolClient(const std::string& protocol,
                                       mate::Arguments* args) {
-  if (protocol.empty())
-    return false;
-
   base::string16 exe;
   if (!GetProtocolLaunchPath(args, &exe))
     return false;
 
+  base::string16 currentClient = GetDefaultProtocolClient(protocol);
+  if (currentClient == L"") {
+    return false;
+  }
+  return currentClient == exe;
```

Comment:
`exe` is guaranteed not empty here, so `return GetDefaultProtocolClient(protocol) == exe` would be simpler;